transcriptLoader.py

```diff
@@ -67,96 +115,140 @@ def validateVideoFiles(self):
+        lines = f.readlines()
+
+    transcript = []
+
+    timeFormat = "%H:%M:%S,%f"
+    arrow = "-->"
+
+    sentence = ""
+    startTime, endTime = "", ""
+
+    for line in lines:
+        line = line.strip()
+        if line.isdigit():
+            continue
+        elif arrow in line:
+            startTime, endTime = line.split(arrow)
+            startTime = datetime.strptime(startTime.strip(), timeFormat)  # .time()
+            endTime = datetime.strptime(endTime.strip(), timeFormat)  # .time()
+        elif line:
+            sentence += " " + line
+        else:
+            transcript.append(
+                {"Line": sentence.strip(), "Start": startTime, "End": endTime}
+            )
+            sentence = ""
+
+    transcriptDF = pd.DataFrame(transcript)
+
+    if transcriptDF.shape[0] == 0:
+        logging.error(f"No transcript data found in {srtFiles[0]}. Exiting...")
+        sys.exit("No transcript data found. Exiting...")
+
+    logging.info(f"Transcript data extracted from {srtFiles[0]}")
+    logging.info(f"Transcript data shape: {transcriptDF.shape}")
+    logging.info(f"Transcript data head: {transcriptDF.head(5)}")
+
+    return transcriptDF
+
+
+def getCombinedTranscripts(transcript, windowSize=30):
+    """
+    Combines overlapping transcripts within a given window size.
+
+    Args:
+        transcript (pandas.DataFrame): The input transcript data.
+        windowSize (int, optional): The window size in seconds. Defaults to 30.
+
+    Returns:
+        pandas.DataFrame: The combined transcript data.
+
+    """
+    transcript = transcript.sort_values(by="Start")
+
+    combinedTranscriptList = []
+    currStart = transcript.iloc[0]["Start"]
+    duration = pd.Timedelta(seconds=windowSize)
+
+    while currStart < transcript.iloc[-1]["Start"]:
+        slicedTranscript = transcript[
+            (transcript["Start"] - currStart < duration)
+            & (transcript["Start"] >= currStart)
+        ]
+
+        if slicedTranscript.shape[0] == 0:
+            duration = pd.Timedelta(seconds=duration.seconds + 1)
+            continue
+
+        combinedLines = " ".join(slicedTranscript["Line"].tolist())
+        combinedTranscriptList.append(
+            {
+                "Combined Lines": combinedLines,
+                "Start": slicedTranscript.iloc[0]["Start"],
+                "End": slicedTranscript.iloc[-1]["End"],
+            }
+        )
+
+        currStart = slicedTranscript.iloc[-1]["End"]
+        duration = pd.Timedelta(seconds=windowSize)
+
+    combinedTranscript = pd.DataFrame(combinedTranscriptList)
+    logging.info(
+        f"Combined Transcript data shape: {combinedTranscript.shape}",
+    )
+    logging.info(f"Combined Transcript data head: {combinedTranscript.head(5)}")
+
+    return combinedTranscript
+
+
+def retrieveTranscript(config, overwrite=False):
+
+    transcriptData = TranscriptData(config)
+    if not config.overwriteTranscriptData and not overwrite:
```

Comment:
What is the purpose of the `overwrite` variable here?  and how it is different from `overwriteTranscriptData`

Reply:
It's purely for debugging purposes. In the Jupyter notebook, I can pass `overwrite` directly in the middle of running other cells without changing the config `.env` value for whatever I might be testing, It has the same functionality as `overwriteTranscriptData`. I use the `overwrite` variable in all of the retrieval functions with the same intention.

Reply:
Can you document that? I was confused and so will next person who look at the code.

Reply:
I have added this in the latest commit to all function that has this parameter, and point hem to use the corresponding `.env` variable instead.